### PR TITLE
Added OAuth2 support.

### PR DESCRIPTION
**Pull request by sfdrogojan**
- Added OAuth2 support.
- Fixed bug caused by accessToken expiration. (https://github.com/salesforce-marketingcloud/FuelSDK-Java/issues/5)

### CODE REVIEW

File: src/main/java/com/exacttarget/fuelsdk/ETClient.java [hunk 1]
```diff
@@ -154,11 +157,13 @@ public ETClient(ETConfiguration configuration)
             gson = gsonBuilder.create();
         }
 
+        useOAuth2Authentication = configuration.isTrue("useOAuth2Authentication") ? true : false;
```
Comment (manivinesh) [inline src/main/java/com/exacttarget/fuelsdk/ETClient.java]: does this error out when the property is not available in config file?

File: src/main/java/com/exacttarget/fuelsdk/ETClient.java [hunk 2]
```diff
@@ -313,6 +322,70 @@ public synchronized String requestToken()
...
+        //
+        // Parse the JSON response into the appropriate instance
+        // variables:
+        //
+
+        String responsePayload = response.getResponsePayload();
+
+        JsonParser jsonParser = new JsonParser();
+        jsonObject = jsonParser.parse(responsePayload).getAsJsonObject();
+        logger.debug("received OAuth2 token:");
```
Comment (manivinesh) [inline src/main/java/com/exacttarget/fuelsdk/ETClient.java]: we should remove these debug logs

File: src/main/java/com/exacttarget/fuelsdk/ETClient.java [hunk 3]
```diff
@@ -418,15 +495,13 @@ public synchronized String refreshToken()
...
-
-            if (refreshToken == null) {
-                throw new ETSdkException("refreshToken == null");
-            }
         }
 
         requestToken(refreshToken);
 
-        soapConnection.setAccessToken(accessToken);
+        if(soapConnection != null) {
```
Comment (manivinesh) [inline src/main/java/com/exacttarget/fuelsdk/ETClient.java]: why we are adding this null check?

File: src/main/java/com/exacttarget/fuelsdk/ETClient.java (hunk 1 above)
Comment (sfdrogojan) [reply src/main/java/com/exacttarget/fuelsdk/ETClient.java]: No. There is a null check for the case when the key is missing from the config file and the method returns false.

File: src/main/java/com/exacttarget/fuelsdk/ETClient.java (hunk 3 above)
Comment (sfdrogojan) [reply src/main/java/com/exacttarget/fuelsdk/ETClient.java]: While doing tests involving accessToken expiration, I encountered a case where this code fails due to the soapConnection being null.

File: src/main/java/com/exacttarget/fuelsdk/ETClient.java (hunk 2 above)
Comment (sfdrogojan) [reply src/main/java/com/exacttarget/fuelsdk/ETClient.java]: Removed debug logs.

**Pull request merged**